lib/gst/player/gstplayer-media-info.c

```diff
@@ -0,0 +1,562 @@
+#include "gstplayer-media-info.h"
+#include "gstplayer-media-info-private.h"
+
+/* Per-stream information */
+G_DEFINE_TYPE (GstPlayerStreamInfo, gst_player_stream_info, G_TYPE_OBJECT);
+
+static void
+gst_player_stream_info_init (GstPlayerStreamInfo * sinfo)
+{
+  /* Nothing to do here */
+}
+
+static void
+gst_player_stream_info_finalize (GObject * object)
+{
+  GstPlayerStreamInfo *sinfo = GST_PLAYER_STREAM_INFO (object);
+
+  if (sinfo->caps) {
+    gst_caps_unref (sinfo->caps);
+    sinfo->caps = NULL;
+  }
+
+  if (sinfo->tags) {
+    gst_tag_list_unref (sinfo->tags);
+    sinfo->tags = NULL;
+  }
+}
+
+static void
+gst_player_stream_info_class_init (GObjectClass * klass)
+{
+  klass->finalize = gst_player_stream_info_finalize;
+}
+
+/**
+ * gst_player_stream_info_get_stream_id:
+ * @info: a #GstPlayerStreamInfo
+ *
+ * Returns: the stream ID of this stream.
+ */
+gint
+gst_player_stream_info_get_stream_id (const GstPlayerStreamInfo *info)
+{
+  return info->stream_id;
```

Comment:
It might be better to call this "stream index", and for stream-id you could use the string-typed stream-id from the -stream-start event of the relevant stream. See gst_event_parse_stream_start()